bundles/mapping/mapmodule/event/MapTourEvent.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * @class Oskari.mapframework.event.common.MapTourEvent
+ *
+ * Notifies application bundles that a map has moved.
+ * See Oskari.mapframework.request.common.MapMoveRequest
+ */
+Oskari.clazz.define('Oskari.mapframework.event.common.MapTourEvent',
+
+    /**
+     * @static @method create called automatically on construction
+     */
+    function (status, location, completed, cancelled) {
+        this._status = status;
+        this._location = location;
+        this._completed = completed;
+        this._cancelled = cancelled;
+    }, {
+        /** @static @property __name event name */
+        __name: 'MapTourEvent',
+
+        /**
+         * @method getName
+         * @return {String} event name
+         */
+        getName: function () {
+            return this.__name;
+        },
+
+        /**
+         * @method getStatus
+         * @return {Object} event status
+         */
+        getStatus: function () {
+            return this._status;
+        },
+
+        /**
+         * @method getLocation
+         * @return {Object} current x and y
+         */
+        getLocation: function () {
+            return this._location;
+        },
+
+        /**
+         * @method getCompleted
+         * @return {Boolean} true if tour is completed
+         */
+        getCompleted: function () {
+            return this._completed;
+        },
+
+        /**
+         * @method getCancelled
+         * @return {Boolean} true if tour is cancelled
+         */
+        getCancelled: function () {
+            return this._cancelled;
+        },
+
+        /**
+         * @method getParams
+         */
+        getParams: function () {
+            return {
+                status: this.getStatus(),
+                location: this.getLocation(),
+                completed: this.getCompleted()
```

Comment:
Not sure if we need canceled or if it can be calculated here from status. But now it's not available for RPC (only in geoportal with the getCancelled() method).

Reply:
I guess status can be calculated with status.steps !== status.step (if the completed step wasn't the last one -> was canceled)?

Reply:
If we add me.notifyTourEvent(status, true) to the !more check, we get cancellation event from tourMap.